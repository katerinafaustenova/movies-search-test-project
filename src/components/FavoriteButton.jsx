import classNames from "classnames";
import React from "react";
import { ReactComponent as Star } from "../assets/star.svg";
import styles from "./FavoriteButton.module.css";

function FavoriteButton({
  id,
  isFavorite = false,
  toggleFavorite,
  positionTop = false,
}) {
  if (!id || !toggleFavorite) return;
  return (
    <button
      onClick={(e) => {
        e.preventDefault();
        toggleFavorite(id);
      }}
      className={classNames(styles.favoriteBtn, {
        [styles.isFavorite]: isFavorite,
        [styles.positionTop]: positionTop,
      })}
    >
      <Star />
    </button>
  );
}

export default FavoriteButton;
