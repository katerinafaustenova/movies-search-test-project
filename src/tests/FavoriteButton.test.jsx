import renderer from "react-test-renderer";
import FavoriteButton from "../components/FavoriteButton";

describe("FavoriteButton", () => {
  const dataProvider = [
    { data: {}, description: "no props" },
    {
      data: {
        isFavorite: false,
        toggleFavorite: jest.fn(),
        positionTop: true,
      },
      description: "no id",
    },
    {
      data: {
        id: "tt12347",
        isFavorite: true,
        toggleFavorite: jest.fn(),
        positionTop: false,
      },
      description: "is favorite and standard position",
    },
    {
      data: {
        id: "tt12348",
        isFavorite: false,
        toggleFavorite: jest.fn(),
        positionTop: true,
      },
      description: "is not favorite and position top",
    },
  ];

  dataProvider.forEach((props) => {
    it(`renders correctly with ${props.description}`, () => {
      const tree = renderer.create(<FavoriteButton {...props.data} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
