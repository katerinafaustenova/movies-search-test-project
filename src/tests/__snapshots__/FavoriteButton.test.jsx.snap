// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteButton renders correctly with is favorite and standard position 1`] = `
<button
  className="favoriteBtn isFavorite"
  onClick={[Function]}
>
  <svg>
    star.svg
  </svg>
</button>
`;

exports[`FavoriteButton renders correctly with is not favorite and position top 1`] = `
<button
  className="favoriteBtn positionTop"
  onClick={[Function]}
>
  <svg>
    star.svg
  </svg>
</button>
`;

exports[`FavoriteButton renders correctly with no id 1`] = `null`;

exports[`FavoriteButton renders correctly with no props 1`] = `null`;
